### ISSUE TITLE
BUG: PRNG key reuse in HPO training loop destroys epoch randomisation

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#47]
## Description

In `optimisation/optimization_train_loop.py`, line 402 resets `train_key = key` at the end of every epoch, where `key` is the original seed from line 122. This means every epoch starts with the **same PRNG state**, producing identical collocation point samples across all epochs.

```python
# Line 122
key = random.PRNGKey(trial_cfg["training"]["seed"])
...
# Line 356 — correctly splits the key
train_key, epoch_key = random.split(train_key)
...
# Line 402 — BUG: resets to original seed
train_key = key
```

## Impact

- Every HPO trial trains on identical data every epoch
- Severely degrades sample diversity and final model quality
- Results from any HPO study using this code are compromised

## Fix

Delete line 402. The split on line 356 already correctly advances the PRNG chain.

## Category
**Critical Bug** · Pillar: Logic Check
